isotpsend.c

```diff
@@ -78,12 +79,32 @@ void print_usage(char *prg)
 	fprintf(stderr, "         -S            (SF broadcast mode - for functional addressing)\n");
 	fprintf(stderr, "         -C            (CF broadcast mode - no wait for flow controls)\n");
 	fprintf(stderr, "         -L <mtu>:<tx_dl>:<tx_flags>  (link layer options for CAN FD)\n");
+	fprintf(stderr, "         -i <filename> (Read PDU data from binary file instead of STDIN)\n");
 	fprintf(stderr, "\nCAN IDs and addresses are given and expected in hexadecimal values.\n");
 	fprintf(stderr, "The pdu data is expected on STDIN in space separated ASCII hex values.\n");
 	fprintf(stderr, "(*) = Use '-t %s' to set N_As to zero for Linux version 5.18+\n", ZERO_STRING);
 	fprintf(stderr, "\n");
 }
 
+ssize_t read_binary_file(void *buffer, const size_t siz_buffer, const char *filename)
+{
+	int fd = -1;
+	ssize_t num_read = -1;
+
+	if( buffer == NULL  ||  siz_buffer < 1  ||  filename == NULL ) {
+		return 0;
+	}
+
+	if( (fd = open(filename, O_RDONLY)) == -1 ) {
```

Comment:
no assignments in if-statements
please adopt the coding style with parenthesis and spaces